compiler/src/dmd/mtype.d

```diff
@@ -4773,16 +4790,96 @@ extern (C++) final class TypeFunction : TypeNext
                 match = m; // pick worst match
         }
 
-        if (pMessage && !parameterList.varargs && nargs > nparams)
+        if (pMessage && !parameterList.varargs && args.length > nparams)
         {
             // all parameters had a match, but there are surplus args
-            *pMessage = getMatchError("expected %d argument(s), not %d", nparams, nargs);
+            *pMessage = getMatchError("expected %d argument(s), not %d", nparams, args.length);
             return MATCH.nomatch;
         }
         //printf("match = %d\n", match);
         return match;
     }
 
+    /********************************
+     * Re-order `argumentList` based on named arguments it contains
+     *
+     * Params:
+     *      argumentList = array of function arguments
+     *      pMessage = address to store error message, or `null`
+     * Returns: re-ordered argument list, or `null` on error
+     */
+    extern(D) Expressions* resolveNamedArgs(ArgumentList argumentList, const(char)** pMessage)
```

Comment:
use `out` for `pMessage`

Reply:
I'm concerned about pMessage being a memory leak

Reply:
It can't be `out` because it can be `null`. It's probably leaking memory, but that's how `callMatch` currently works in order to give better error messages for template functions.